### ISSUE TITLE
Fail to install via hyprpm

### ISSUE THREAD

**Issue by winterSteve25** [Duckonaut/split-monitor-workspaces#50]
this error is emitted when running `hyprpm add https://github.com/Duckonaut/split-monitor-workspaces -v`

```
The Meson build system
Version: 1.3.2
Source dir: /tmp/hyprpm/new
Build dir: /tmp/hyprpm/new/build
Build type: native build
Project name: split-monitor-workspaces
Project version: 0.1
C++ compiler for the host machine: c++ (gcc 13.2.1 "c++ (GCC) 13.2.1 20230801")
C++ linker for the host machine: c++ ld.bfd 2.42.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments -std=c++23: YES 
meson.build:8: WARNING: Consider using the built-in option for language standard version instead of using "-std=c++23".
Found pkg-config: YES (/usr/bin/pkg-config) 2.1.0
Run-time dependency hyprland found: YES 0.35.0
Run-time dependency pixman-1 found: YES 0.43.2
Run-time dependency libdrm found: YES 2.4.120
Build targets in project: 1

Found ninja-1.11.1 at /usr/bin/ninja
                                                
 -> cd /tmp/hyprpm/new && PKG_CONFIG_PATH="/home/cadenz/.local/share/hyprpm/headersRoot/share/pkgconfig" meson compile -C build
ninja: Entering directory `/tmp/hyprpm/new/build'
[1/2] Compiling C++ object libsplit-monitor-workspaces.so.p/src_main.cpp.o
FAILED: libsplit-monitor-workspaces.so.p/src_main.cpp.o 
c++ -Ilibsplit-monitor-workspaces.so.p -I. -I.. -I../include -I/home/cadenz/.local/share/hyprpm/headersRoot/include/hyprland/protocols -I/home/cadenz/.local/share/hyprpm/headersRoot/include/hyprland/wlroots -I/home/cadenz/.local/share/hyprpm/headersRoot/include -I/usr/include/pixman-1 -I/usr/include/libdrm -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -std=c++23 -DWLR_USE_UNSTABLE -fPIC -MD -MQ libsplit-monitor-workspaces.so.p/src_main.cpp.o -MF libsplit-monitor-workspaces.so.p/src_main.cpp.o.d -o libsplit-monitor-workspaces.so.p/src_main.cpp.o -c ../src/main.cpp
../src/main.cpp: In function ‘void mapWorkspacesToMonitors()’:
../src/main.cpp:125:45: error: ‘Hyprlang’ has not been declared
  125 |     static auto* const workspaceCountPtr = (Hyprlang::INT* const*)HyprlandAPI::getConfigValue(PHANDLE, k_workspaceCount)->getDataStaticPtr();
      |                                             ^~~~~~~~
../src/main.cpp:125:60: error: expected primary-expression before ‘const’
  125 |     static auto* const workspaceCountPtr = (Hyprlang::INT* const*)HyprlandAPI::getConfigValue(PHANDLE, k_workspaceCount)->getDataStaticPtr();
      |                                                            ^~~~~
../src/main.cpp:125:59: error: expected ‘)’ before ‘const’
  125 |     static auto* const workspaceCountPtr = (Hyprlang::INT* const*)HyprlandAPI::getConfigValue(PHANDLE, k_workspaceCount)->getDataStaticPtr();
      |                                            ~              ^~~~~~
      |                                                           )
../src/main.cpp:126:42: error: ‘Hyprlang’ has not been declared
  126 |     static auto* const keepFocusedPtr = (Hyprlang::INT* const*)HyprlandAPI::getConfigValue(PHANDLE, k_keepFocused)->getDataStaticPtr();
      |                                          ^~~~~~~~
../src/main.cpp:126:57: error: expected primary-expression before ‘const’
  126 |     static auto* const keepFocusedPtr = (Hyprlang::INT* const*)HyprlandAPI::getConfigValue(PHANDLE, k_keepFocused)->getDataStaticPtr();
      |                                                         ^~~~~
../src/main.cpp:126:56: error: expected ‘)’ before ‘const’
  126 |     static auto* const keepFocusedPtr = (Hyprlang::INT* const*)HyprlandAPI::getConfigValue(PHANDLE, k_keepFocused)->getDataStaticPtr();
      |                                         ~              ^~~~~~
      |                                                        )
../src/main.cpp: In function ‘PLUGIN_DESCRIPTION_INFO pluginInit(void*)’:
../src/main.cpp:170:60: error: ‘Hyprlang’ has not been declared
  170 |     HyprlandAPI::addConfigValue(PHANDLE, k_workspaceCount, Hyprlang::INT{10});
      |                                                            ^~~~~~~~
../src/main.cpp:171:57: error: ‘Hyprlang’ has not been declared
  171 |     HyprlandAPI::addConfigValue(PHANDLE, k_keepFocused, Hyprlang::INT{0});
      |                                                         ^~~~~~~~
ninja: build stopped: subcommand failed.
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /usr/bin/ninja -C /tmp/hyprpm/new/build
```

Hyprland version: 0.35.0-2

**Comment by winterSteve25** [comment]
I forked the repo and reverted the Hyprlang v4 changes and it seems to work.. maybe its outdated headers on my side?

**Comment by liferooter** [comment]
The problem is reproduced on NixOS with Hyprland 0.35

**Comment by shangqoengsuang** [comment]
same here, Archlinux with Hyprland 0.35

**Comment by kofteistkofte** [comment]
I have the same issue on Arch Linux with Hyprland 0.35

**Comment by kofteistkofte** [comment]
It seems like the issue only exists on Hyprland 0.35 and before.  I updated to Hyprland 0.36 and it builds now. The changes in #48 expects the changes in Hyprland 0.36. For example `HyprlandAPI::getConfigValue` returns `SConfigValue` in 0.35 but returns `Hyprlang::CConfigValue` in 0.36.

In my understanding, the issue will fix itself when distros update to Hyprland 0.36, but I don't know if the author should implement a temporary fix for this.

**Comment by Duckonaut** [comment]
Nope, unfortunately, there's no real way to have a version with both a pre-0.36 and post-0.36 implementation in a not-stupid way. The changes were introduced due to Hyprland changing its configuration to a new system, and won't work on versions from before then.
